src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -66,15 +70,42 @@ createSQLiteStore dbFilePath = do
   compileOptions <- liftIO (DB.query_ (dbConn store) "pragma COMPILE_OPTIONS;" :: IO [[T.Text]])
   let threadsafeOption = find (isPrefixOf "THREADSAFE=") (concat compileOptions)
   liftIO $ case threadsafeOption of
-    Just "THREADSAFE=0" -> do
-      putStrLn "SQLite compiled with not threadsafe code, continue (y/n):"
-      s <- getLine
-      when (s /= "y") (exitWith $ ExitFailure 2)
+    Just "THREADSAFE=0" -> confirmOrExit "SQLite compiled with non-threadsafe code."
     Nothing -> putStrLn "Warning: SQLite THREADSAFE compile option not found"
     _ -> return ()
-  liftIO . createSchema $ dbConn store
+  liftIO $ migrateSchema store
   return store
 
+migrateSchema :: SQLiteStore -> IO ()
+migrateSchema SQLiteStore {dbConn, dbFilePath} = do
+  Migrations.initialize dbConn
+  Migrations.get dbConn Migrations.app >>= \case
+    NoMigration -> pure ()
+    MigrateError e -> confirmOrExit $ "Database is possibly corrupted: " <> e
+    ms@MigrateDown {} -> do
+      confirmOrExit
+        "The chat database has a newer version than the app.\n\
+        \It is recommended to run a newer version of the app.\n\
+        \If you continue the database will be backed up and downgraded."
+      migrate ms
+    ms@MigrateUp {dbMigrations} -> do
+      unless (null dbMigrations) $
+        confirmOrExit "The app has a newer version that the chat database - it will be backed up and upgraded."
```

Comment:
that -> than

Reply:
you can't back up the app, so unnecessary clarification